dolphin/api/v1/storages.py

```diff
@@ -123,8 +132,31 @@ def create(self, req, body):
 
         return storage_view.build_storage(storage)
 
+    @validation.schema(schema_storages.update)
     def update(self, req, id, body):
-        return dict(name="Storage 4")
+        """Update storage access information."""
+        ctx = req.environ.get('dolphin.context')
+        try:
+            # Get existing access_info from DB
+            access_info_dict = db.access_info_get(ctx, id)
+            # Merge fields from request body to access_info
+            update_access_info(access_info_dict, body)
+            # Discover Storage with new access info
+            storage = self.driver_manager.get_storage(ctx, access_info_dict['storage_id'])
+            # Need to encode password before updating to DB
+            body['password'] = cryptor.encode(access_info_dict['password'])
+            db.access_info_update(ctx, id, body)
```

Comment:
why not use access_info here?

Reply:
I think you can do the check and update of password update [line#147] in the update_Access_info function and db.access_info_update should be passed with accee_info_dict

Reply:
done